graphql_api/helpers/connection.py

```diff
@@ -68,6 +69,112 @@ def page_info(self, *args, **kwargs):
         }
 
 
+class ArrayPaginator:
+    """Cursor-based paginator for in-memory arrays."""
```

Comment:
I was debating between using stringified random cursors and numeric ones for pagination. I decided to go with numeric cursors because they are simple, efficient, and directly map to the array indices (we have cases where customers have over 1,000 files), making pagination faster and easier to debug. Since the dataset is static, there’s no need for the added complexity or security of random strings. If you have other perspectives, lmk